pytorch_lightning/accelerators/accelerator.py

```diff
@@ -202,6 +202,21 @@ def sync_tensor(self,
         """
         raise NotImplementedError()
 
+    @property
+    def optimizer_states(self) -> List[dict]:
+        """
+        Returns states of optimizers. Allows for syncing/collating optimizer state from processes in custom
+        plugins.
+        Return:
+            Optimizer state dicts
+        """
+        if self.ddp_plugin:
+            return self.ddp_plugin.optimizer_states(self.trainer)
```

Comment:
does this need to go through the DDP plugin? could the base accelerator define this as L215-218, and then the sharded ddp accelerator overrides `optimizer_states` more directly?

Reply:
The sharded ddp accelerator is now a plugin (will update the original PR title), so all the sharded ddp logic lives within a plugin that is then put into each accelerator.